lib/core/mergeConfig.js

```diff
@@ -56,9 +56,10 @@ module.exports = function mergeConfig(config1, config2) {
     .concat(defaultToConfig2Keys);
 
   var otherKeys = Object
-    .keys(config2)
-    .filter(function filterAxiosKeys(key) {
-      return axiosKeys.indexOf(key) === -1;
+    .keys(config1)
+    .concat(Object.keys(config2))
+    .filter(function filterAxiosKeys(key, index, self) {
+      return axiosKeys.indexOf(key) === -1 && self.indexOf(key) === index;
```

Comment:
I am curious about the meaning of `self.indexOf(key) === index`.

Reply:
+1 - as I understand it, the real fix here is concatenating the keys of the default config options and the request level options, as previously this code wasn't picking up default config options, not sure what the bonus `self.indexOf(key)` does :)